app/controllers/api/v1/visualization/expression_controller.rb

```diff
@@ -159,6 +160,17 @@ def render_heatmap
 
           render plain: expression_data, status: 200
         end
+
+        # enforce a limit on number of genes allowed for visualization requests
+        # see StudySearchService::MAX_GENE_SEARCH
+        def check_gene_limit
+          return true if params[:genes].blank?
+
+          # render 422 if more than MAX_GENE_SEARCH as request fails internal validation
+          if params[:genes].split(',').size > StudySearchService::MAX_GENE_SEARCH
+            render json: { error: StudySearchService::MAX_GENE_SEARCH_MSG }, status: :unprocessable_entity
```

Comment:
Per other analytics suggestion:

```suggestion
          num_genes = params[:genes].split(',').size
          if num_genes > StudySearchService::MAX_GENE_SEARCH
            MetricsService.log('search-too-many-genes', {numGenes: num_genes})
            render json: { error: StudySearchService::MAX_GENE_SEARCH_MSG }, status: :unprocessable_entity
```